quic/s2n-quic-rustls/src/session.rs

```diff
@@ -255,6 +274,11 @@ impl tls::Session for Session {
                         }
                         quic::KeyChange::OneRtt { keys, next } => {
                             let (key, header_key) = OneRttKey::new(keys, next, cipher_suite);
+
+                            if let Connection::Client(_) = &self.connection {
```

Comment:
Same critique as [this comment](https://github.com/aws/s2n-quic/pull/1238/files#r833764775), the client should have read the server hello before calculating handshake keys, so I think we can emit this value where the server does(line 264).

Reply:
see discussion above: https://github.com/aws/s2n-quic/pull/1238#discussion_r833812067